File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int>st;
        for(int i =0 ; i< nums.size(); i++){
            if(!st.empty()){
                if(st.find(nums[i]) != st.end()){
                    return true;
                }
                else{
                    st.insert(nums[i]);
                }
            }else{
                st.insert(nums[i]);
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        n = len(nums)

        for i in range (n):
            val = target - nums[i]

            if val in mp:
                return [mp[val],i]
            
            mp[nums[i]] = i;

        return []